improver/utilities/ancillary_creation.py

```diff
@@ -158,87 +192,105 @@ def unnormalised_smoothing_coefficients(self, gradient_cube):
             iris.cube.Cube:
                 The cube of initial unscaled smoothing_coefficients
         """
-        smoothing_coefficients_cube = gradient_cube.copy(
-            data=self.coefficient * gradient_cube.data ** self.power
-        )
-        return smoothing_coefficients_cube
+        return self.coefficient * gradient_cube.data ** self.power
 
     @staticmethod
-    def update_smoothing_coefficients_metadata(smoothing_coefficients_cube, cube_name):
+    def create_coefficient_cube(data, template, cube_name, attributes):
         """
         Update metadata in smoothing_coefficients cube. Remove any time
         coordinates and rename.
 
         Args:
-            smoothing_coefficients_cube (iris.cube.Cube):
-                A cube of smoothing_coefficients with "gradient" metadata
+            data (numpy.array):
+                The smoothin coefficient data to store in the cube.
+            template (iris.cube.Cube):
+                A gradient cube, the dimensions of which are used as a template
+                for the coefficient cube.
             cube_name (str):
                 A name for the resultant cube
+            attributes (dict):
+                A dictionary of attributes for the new cube.
 
         Returns:
             iris.cube.Cube:
-                A cube of smoothing_coefficients with adjusted metadata
+                A new cube of smoothing_coefficients
         """
-        smoothing_coefficients_cube.rename(cube_name)
-        for coord in smoothing_coefficients_cube.coords(dim_coords=False):
-            if "time" in coord.name() or "period" in coord.name():
-                smoothing_coefficients_cube.remove_coord(coord)
-        return smoothing_coefficients_cube
+        for coord in template.coords(dim_coords=False):
+            for coord_name in ["time", "period", "realization"]:
```

Comment:
Primarily because if a slice of the multi-realization ENUKX orography (for example) is provided it will contain a scalar realization coordinate which has no value in the output smoothing coefficients field.

Reply:
Ok